pkg/instrumentation/nethttp/client/client_hook.go

```diff
@@ -0,0 +1,153 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/open-telemetry/opentelemetry-go-compile-instrumentation/pkg/inst"
+	"github.com/open-telemetry/opentelemetry-go-compile-instrumentation/pkg/instrumentation/nethttp/semconv"
+	"github.com/open-telemetry/opentelemetry-go-compile-instrumentation/pkg/instrumentation/shared"
+)
+
+const (
+	otelExporterPrefix     = "OTel OTLP Exporter Go"
+	instrumentationName    = "github.com/open-telemetry/opentelemetry-go-compile-instrumentation/pkg/instrumentation/nethttp"
+	instrumentationVersion = "0.1.0"
+)
+
+var (
+	logger     = shared.GetLogger()
+	tracer     trace.Tracer
+	propagator propagation.TextMapPropagator
+	initOnce   sync.Once
+)
+
+func initInstrumentation() {
+	initOnce.Do(func() {
+		if err := shared.SetupOTelSDK(); err != nil {
+			logger.Error("failed to setup OTel SDK", "error", err)
+		}
+		tracer = otel.GetTracerProvider().Tracer(
+			instrumentationName,
+			trace.WithInstrumentationVersion(instrumentationVersion),
+		)
+		propagator = otel.GetTextMapPropagator()
+		logger.Info("HTTP client instrumentation initialized")
+	})
+}
+
+// netHttpClientEnabler controls whether client instrumentation is enabled
+type netHttpClientEnabler struct{}
+
+func (n netHttpClientEnabler) Enable() bool {
+	return shared.Instrumented("NETHTTP")
```

Comment:
I see this `NETHTTP` literal is used several times, good candidate for const, to be more safe